### ISSUE TITLE
Add mo3 tag based browsing

### ISSUE THREAD

**Issue by alirz1** [AndreyPavlenko/Fermata#3]
While organized folders is a good way to browse for music, I think mp3 tag based, at least ARTIST,ALBUM,SONG name based browsing should be added. 
Along with a global search  option. That would make this perfect.Keeping the current interface that is simple and excellent.

Thanks.

**Comment by AndreyPavlenko** [comment]
Added to my TODO list.

**Comment by alirz1** [comment]
Thanks. I’m thinking something that navigates like the old iPod style navigation. 
Right where you have the go back button. There could be a global search button on top of that. Just like the back button remains on all pages, the search button would also remain on all pages, making easier to quickly search for an artists, album or song in the folder.

I hope you find these tips useful and you find them inline with your design thought for the music player. There are too many android music players out there that have complex GUI based interfaces and a useless in auto mode. A simple text based menu like yours is what’s we need.

**Comment by AndreyPavlenko** [comment]
To search (by title) in the current folder there is the filter button on top, in the toolbar. I can extend its functionality to filter by album, artist etc.

I was thinking to implement something like search folders. Just like currently we have the Playlists tab, where we can add/modify playlists, we may have a separate Search tab, where we could create search folders. As an example:
Folder name: New Classic
Search: Genre=Classic && Year=current
Sort by: Date, descending
Group by: Composer

Every time, clicking on this New Classic folder, we will see only the items corresponding to the specified search criteria.

**Comment by alirz1** [comment]
Honestly I personally dislike the search tab. Every other music app out there has it and I find it a bad implementation, going through tabs is distracting. Again that's my personal opinion. I'm not a programmer thus I can only give my input as an end user.
I've tried thousands of music apps and you're the first one that I've come across that is on GitHub and whose app seems to be almost what I was looking for. While I love your app as is, I just wanted to give some input as to what I think would make it perfect.
Honestly, in my opinion the old iPod style interface is perfect for this use. It's simple, clutter free. You click on one menu to get to the next. Add on a search button always available on the right side above the back button and done. The search button can either look under the current folder for the mp3 tags or filename based search, which ever you want and is easier. 
Curious, have you used the old iPods from apple?

**Comment by alirz1** [comment]
If you don't mind me saying  going to system menus and chexking boxes to enable/disable search type etc is very distracting, especially if we are looking to use this in car mode.

**Comment by AndreyPavlenko** [comment]
Perhaps my description was not clear... I didn't mean an ordinary search tab. Think about it like that it's a Playlists tab. But as distinct from the Playlists tab, you add there search queries, but not particular tracks. Another example - "My Favorite Singer".  Whenever you click on this search folder, you get the list of your Favorite Singer songs found in the music folders. If you download new songs to your device, you'll see them in the list.

What you are talking about (in my understanding) is a search in the current folder. This functionality is currently implemented as a filter. Currently it's not possible to filter by artist, album, etc., only by title, which is either track or file name.

No, I've never used an iPod.

Anyway, thanks a lot for your feedback.

**Comment by alirz1** [comment]
Thanks. How about also a text/font size option? Would that be hard to add?

**Comment by AndreyPavlenko** [comment]
No, that's not hard. Added to my list. You may also create a separate issue for tracking.
